Sources/Foundation/Host.swift

```diff
@@ -25,7 +25,8 @@ import WinSDK
 
     // getnameinfo uses size_t for its 4th and 6th arguments.
     private func getnameinfo(_ addr: UnsafePointer<sockaddr>?, _ addrlen: socklen_t, _ host: UnsafeMutablePointer<Int8>?, _ hostlen: socklen_t, _ serv: UnsafeMutablePointer<Int8>?, _ servlen: socklen_t, _ flags: Int32) -> Int32 {
-        return Glibc.getnameinfo(addr, addrlen, host, Int(hostlen), serv, Int(servlen), flags)
+        guard let saddr = addr else { return -1 }
```

Comment:
I think that I prefer my approach of changing the signature of the private func.

Reply:
Sorry about the delay, been meaning to get to this, hopefully this week.

Reply:
Changing the signature would work too, dropped this in #5010, you can submit it in a version of #4889 for 6.0.